dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/action/EditContentletAction.java

```diff
@@ -628,12 +643,20 @@ else if ((cmd != null) && cmd.equals(com.dotmarketing.util.Constants.UNLOCK)) {
 			_sendToReferral(req, res, referer);
 		}else if(cmd != null && cmd.equals("export")){
 			try {
-				String structureInode = req.getParameter("expStructureInode");
-
-				ActionResponseImpl resImpl = (ActionResponseImpl) res;
-				HttpServletResponse response = resImpl.getHttpServletResponse();
-
-				downloadToExcel(response, user,searchContentlets(req,res,config,form,user,"Excel"), structureInode);
+				final String contentTypeId = req.getParameter("expStructureInode");

```

Comment:
Hate to say it but this needs to be moved to a resource - even as is with a bunch of ugly parameters. We should not invest any more time in Struts actions and should take the opportunity to remove them when possible.

Reply:
Totally agree.
The effort is important so do it right